src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/MacBundlingEnvironment.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.jpackage.internal;
+
+import static jdk.jpackage.internal.MacFromOptions.createMacApplication;
+import static jdk.jpackage.internal.MacPackagingPipeline.APPLICATION_LAYOUT;
+import static jdk.jpackage.internal.MacPackagingPipeline.createSignAppImagePackage;
+import static jdk.jpackage.internal.cli.StandardBundlingOperation.CREATE_MAC_APP_IMAGE;
+import static jdk.jpackage.internal.cli.StandardBundlingOperation.CREATE_MAC_DMG;
+import static jdk.jpackage.internal.cli.StandardBundlingOperation.CREATE_MAC_PKG;
+import static jdk.jpackage.internal.cli.StandardBundlingOperation.SIGN_MAC_APP_IMAGE;
+
+import java.util.Optional;
+import jdk.jpackage.internal.cli.Options;
+import jdk.jpackage.internal.model.MacPackage;
+import jdk.jpackage.internal.model.Package;
+import jdk.jpackage.internal.util.Result;
+
+public class MacBundlingEnvironment extends DefaultBundlingEnvironment {
+
+    public MacBundlingEnvironment() {
+        super(build()
+                .defaultOperation(CREATE_MAC_DMG)
+                .bundler(SIGN_MAC_APP_IMAGE, MacBundlingEnvironment::signAppImage)
+                .bundler(CREATE_MAC_APP_IMAGE, MacBundlingEnvironment::createAppImage)
+                .bundler(CREATE_MAC_DMG, LazyLoad::dmgSysEnv, MacBundlingEnvironment::createDmdPackage)
+                .bundler(CREATE_MAC_PKG, MacBundlingEnvironment::createPkgPackage));
```

Comment:
Do you know why `CREATE_MAC_PKG` does not need/use environment like `CREATE_MAC_DMG`?

Reply:
For dmg packaging, we validate that hdiutil and osascript commands are available. We also need to find the SetFile utility that may be in multiple locations. For pkg, we assume pkgbuild and productbuild commands are available and don't do any validation. That is why we have [MacDmgSystemEnvironment](https://github.com/openjdk/jdk/blob/10f262a6ad9a6e89cd79409c5e1a3f7efda76928/src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/MacDmgSystemEnvironment.java) but don't have MacPkgSystemEnvironment. 

Historically, pkg tools were not validated. We can change it in a separate CR.

Reply:
Thanks for explanation. Lets keep it this way.